src/Nowruz/modules/settings/components/password/usePassword.tsx

```diff
@@ -0,0 +1,54 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEffect, useState } from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { changePassword } from 'src/core/api';
+import { passwordPattern } from 'src/core/regexs';
+import * as yup from 'yup';
+
+import { Inputs } from "./password.type";
+
+const schema = yup.object().shape({
+    current_password: yup.string().required(),
+    password: yup.string().required(),
+    confirm: yup.string().required(),
+});
+
+export const usePassword = () => {
+    const [isPasswordLengthValid, setIsPasswordLengthValid] = useState(false);
+    const [isPasswordPatternValid, setIsPasswordPatternValid] = useState(false);
+    const [isFormValid, setIsFormValid] = useState(false);
+    const [isPasswordMatch, setPasswordMatch] = useState(true);
+    const {
+        watch,
+        register,
+        handleSubmit,
+        formState: { errors },
+        reset,
+    } = useForm<Inputs>({
+        resolver: yupResolver(schema),
+    });
+    const current_password = watch('current_password');
+    const password = watch('password');
+    const confirmPassword = watch('confirm');
+    useEffect(() => {
+        setIsFormValid(
+            !!password && password.length >= 8 && passwordPattern.test(password) && (password === confirmPassword),
+        );
+        setIsPasswordLengthValid(!!(password && password.length >= 8));
```

Comment:
No need to add these states for validation, Yup has built in functions which can manage validations. for example using regex for password pattern. You can use chatGPT to generate regex and use it in YUP. Also for length and repeatPass you can use YUP